ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -290,9 +290,9 @@ local_TIME_WAIT_TIMEOUT = 0
 
 
 socketAddressType :: Socket.SockAddr -> Maybe AddressType
-socketAddressType Socket.SockAddrInet {}  = Just IPv4Address
-socketAddressType Socket.SockAddrInet6 {} = Just IPv6Address
-socketAddressType addr                    =
+socketAddressType Socket.SockAddrInet {}      = Just IPv4Address
+socketAddressType Socket.SockAddrInet6 {}     = Just IPv6Address
+socketAddressType addr@Socket.SockAddrUnix {} =
   error ("socketAddressType: unexpected address " ++ show addr)
```

Comment:
Can this function return `Nothing` instead of erroring when it is provided a `SockAddrUnix`?